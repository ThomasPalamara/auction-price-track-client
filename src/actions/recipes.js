export const selectRecipe = (recipe = {}) => (
  {
    type: 'SELECT_RECIPE',
    recipe,
  });


const defaultRecipes = [{
  professions: [''],
  type: '',
  craft: {
    blizzardId: 0,
    quantity: 1,
  },
  reagents: [{
    blizzardId: 0,
    quantity: 0,
  }, {
    blizzardId: 1,
    quantity: 0,
  }],
}];

export const addRecipes = (recipes = defaultRecipes) => (
  {
    type: 'ADD_RECIPES',
    recipes,
  }
);
