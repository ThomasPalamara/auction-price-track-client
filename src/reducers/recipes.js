
export default (state = [], action) => {
    switch (action.type) {
        case 'ADD_RECIPES':
            return action.recipes;
        case 'SELECT_RECIPE':
            return {...state, selectedRecipe: action.recipe};
        default:
            return state;
    }
}

