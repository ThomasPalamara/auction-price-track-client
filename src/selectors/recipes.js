export default (list, { text, profession, isCustom}) => {
	console.log(list);
	return list.filter((item) => {
		// const textMatch = item.description.toLowerCase().includes(text.toLowerCase());
        const professionMatch = profession !== 'all' ? item.professions.indexOf(profession) != -1 : true ;
        const isCustomMatch = item.isCustom ? item.isCustom === isCustom : false === isCustom;
		return isCustomMatch && professionMatch;
	});
};
