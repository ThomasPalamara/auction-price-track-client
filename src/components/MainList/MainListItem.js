import React from 'react';
import { connect } from 'react-redux'
import { selectRecipe } from '../../actions/recipes'
import WHLink from '../WHLink'

class MainListItem extends React.Component {
    render() {
        return (
            <li
                className={`recipe list-item ${this.props.active ? 'active' : ''} `}
                onClick={() => this.props.dispatch(selectRecipe(this.props.element))}
            >
                <div className={this.props.element.isCustom ? 'generalBg' : this.props.element.professions[0] + 'Bg'}>
                    {this.props.element.craft &&
                        <h6>
                            <WHLink {...this.props.element.craft}/>
                        </h6>
                    }

                    <ul>
                        {this.props.element.reagents.map((reagent) => (
                            <li key={reagent.blizzardId}>
                                <WHLink {...reagent}/>
                            </li>
                        ))
                        }
                    </ul>
                </div>
            </li>)
    }
};
  
export default connect()(MainListItem);