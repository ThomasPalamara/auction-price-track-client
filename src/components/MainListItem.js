import React from 'react';
import { selectRecipe } from '../actions/recipes'

class MainListItem extends React.Component {
    componentDidUpdate(){
    }
    render() {
        return (
            <li
                className={`recipe list-item ${this.props.active ? 'active' : ''} `}
                onClick={() => this.props.dispatch(selectRecipe(this.props.element))}
            >
                <div className={this.props.element.isCustom ? 'generalBg' : this.props.element.professions[0] + 'Bg'}>
                    {this.props.element.craft &&
                        <h6>
                            <a
                                href="javascript:(void)"
                                data-wowhead={`item=${this.props.element.craft.blizzardId}`}
                            >{this.props.element.craft.name}</a>
                        </h6>
                    }

                    <ul>
                        {this.props.element.reagents.map((reagent) => (
                            <li key={reagent.blizzardId}>
                                <a
                                    href="javascript:(void)"
                                    data-wowhead={`item=${reagent.blizzardId}`}
                                >{reagent.name}</a>
                            </li>
                        ))
                        }
                    </ul>
                </div>
            </li>)
    }
};

export default MainListItem;