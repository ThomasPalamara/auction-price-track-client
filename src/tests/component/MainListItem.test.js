import React from 'react';
import { shallow } from "enzyme";
import toJSON from 'enzyme-to-json';
import MainListItem from '../../components/MainListItem';
import recipes from '../fixtures/recipes';

test('Should render MainListItem component correctly for recipe (isCustom === false)', () => {
    const wrapper = shallow(<MainListItem element={recipes[1]}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h6').length).toBe(1);
})

test('Should render MainListItem component correctly for reagent (isCustom === true)', () => {
    const wrapper = shallow(<MainListItem element={recipes[0]}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h6').length).toBe(0);
})
