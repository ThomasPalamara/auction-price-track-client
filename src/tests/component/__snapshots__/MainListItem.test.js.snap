// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MainListItem component correctly for reagent (isCustom === true) 1`] = `
<li
  className="recipe list-item"
  onClick={[Function]}
>
  <div
    className="alchemyBg"
  >
    <ul>
      <li
        key="152510"
      >
        <a
          data-wowhead="item=152510"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152507"
      >
        <a
          data-wowhead="item=152507"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152511"
      >
        <a
          data-wowhead="item=152511"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152508"
      >
        <a
          data-wowhead="item=152508"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152505"
      >
        <a
          data-wowhead="item=152505"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152509"
      >
        <a
          data-wowhead="item=152509"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152506"
      >
        <a
          data-wowhead="item=152506"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
    </ul>
  </div>
</li>
`;

exports[`Should render MainListItem component correctly for recipe (isCustom === false) 1`] = `
<li
  className="recipe list-item"
  onClick={[Function]}
>
  <div
    className="alchemyBg"
  >
    <h6>
      <a
        data-wowhead="item=152641"
        href="javascript:(void)"
      >
        item
      </a>
    </h6>
    <ul>
      <li
        key="152510"
      >
        <a
          data-wowhead="item=152510"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152507"
      >
        <a
          data-wowhead="item=152507"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
      <li
        key="152509"
      >
        <a
          data-wowhead="item=152509"
          href="javascript:(void)"
        >
          item
        </a>
      </li>
    </ul>
  </div>
</li>
`;
